packages/fxa-settings/src/pages/ResetPassword/index.tsx

```diff
@@ -0,0 +1,204 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Localized, useLocalization } from '@fluent/react';
+import { RouteComponentProps } from '@reach/router';
+import React, { useCallback, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useNavigate } from '@reach/router';
+import { HomePath } from '../../constants';
+import {
+  logViewEvent,
+  settingsViewName,
+  usePageViewEvent,
+} from '../../lib/metrics';
+import { useAccount, useAlertBar } from '../../models';
+import { InputText } from '../../components/Settings/InputText';
+import HeaderContinueTo from '../../components/App/HeaderContinueTo';
+import WarningMessage from '../../components/App/WarningMessage';
+import RememberPassword from '../../components/App/RememberPassword';
+
+// --forceEmail-- is a hint to the signup page that the user should not
+// be given the option to change their address
+// can be triggered by clicking on "Forgot password?" from /force_auth?email=user@domain.com
+// where the provided email is a registered account
+// (e.g., if coming from force_auth or a relying party)
+
+// --canGoBack-- determines if the user can navigate back to an fxa entrypoint
+// unsure when this might be false
+
+// --serviceName-- is the relying party
+
+// '?email=username@domain.com&reset_password_confirm=false' query params
+// skips reset_password (and warnings) and goes directly to confirm_reset_password
+
+// TODO verify if these are passed as props or obtained a different way
+export type ResetPasswordProps = {
+  serviceName?: string;
+  forceEmail?: string;
+  canGoBack?: boolean;
+};
+
+type FormData = {
+  email: string;
+};
+
+// eslint-disable-next-line no-empty-pattern
+const ResetPassword = ({
+  serviceName,
+  forceEmail,
+  canGoBack,
+}: ResetPasswordProps & RouteComponentProps) => {
+  usePageViewEvent('settings.reset-password', {
+    entrypoint_variation: 'react',
+  });
+
+  const [email, setEmail] = useState<string>(forceEmail || '');
+  const [emailErrorText, setEmailErrorText] = useState<string>();
```

Comment:
You could initialize this with an empty string so it's always a string instead of a string/undefined (probably what I'd do), or initialize it with `undefined` so it's intentional.